File: backend/Apex/Apex.Domain/Entities/Instrument.cs
using Apex.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace Apex.Domain.Entities
{
    public class Instrument
    {
        public Guid InstrumentId { get; set; }
        public string Symbol { get; set; }
        public string InstrumentName { get; set; }

        public decimal PointValue { get; set; }
        public decimal TickSize { get; set; }

        public decimal TickValue { get; set; }
        public string Currency { get; set; }
        public InstrumentType Type { get; set; }
    }
}

File: frontend/src/types/instrument.ts
// Global, read-only instrument catalog (#94). Seeded server-side and not editable
// by the user, so it can be cached aggressively on the client.
export type InstrumentType = 'Future' | 'Cfd'

export interface InstrumentDto {
  instrumentId: string
  symbol: string
  instrumentName: string
  // Currency per point — what makes P&L money instead of points (#94).
  pointValue: number
  tickSize: number
  tickValue: number
  currency: string
  type: InstrumentType
}
